documentation/docs/05-load.md

```diff
@@ -0,0 +1,338 @@
+	const { a, b } = await parent();
+	console.log(a, b); // `1`, `2`
+
+	return { c: 3 };
+}
+```
+
+In `+page.js` or `+layout.js` it will return data from `load` functions in parent `+layout.js` files. Implicitly, a missing `+layout.js` is treated as a `({ data }) => data` function, meaning that it will also return data from parent `+layout.server.js` files.
+
+#### setHeaders
+
+If you need to set headers for the response, you can do so using the `setHeaders` method. This is useful if you want to cache the response:
+
+```js
+/// file: src/routes/blog/+page.js
+
+// @filename: $types.d.ts
+export type PageLoad = import('@sveltejs/kit').Load<{}>;
+
+// @filename: index.js
+// ---cut---
+/** @type {import('./$types').PageLoad} */
+export async function load({ fetch, setHeaders }) {
+	const url = `https://cms.example.com/articles.json`;
+	const response = await fetch(url);
+
+	setHeaders({
+		'age': response.headers.get('age'),
+		'cache-control': response.headers.get('cache-control')
+	});
+
+	return response.json();
+}
+```
+
+### Output
+
+Any promises on the returned `data` object will be resolved, if they are top-level properties. This makes it easy to return multiple promises without creating a waterfall:
+
+```js
+// @filename: $types.d.ts
+export type PageLoad = import('@sveltejs/kit').Load<{}>;
+
+// @filename: index.js
+// ---cut---
+/** @type {import('./$types').PageLoad} */
+export function load() {
+	return {
+		a: Promise.resolve('a'),
+		b: Promise.resolve('b'),
+		c: {
+			value: Promise.resolve('c')
+		}
+	};
+}
+```
+
+```svelte
+<script>
+	/** @type {import('./$types').PageData} */
+	export let data;
+
+	console.log(data.a); // 'a'
+	console.log(data.b); // 'b'
+	console.log(data.c.value); // `Promise {...}`
+</script>
+```
+
+### Errors
+
+If an error is thrown during `load`, the nearest [`+error.svelte`](/docs/routing#error) will be rendered. For _expected_ errors, use the `error` helper from `@sveltejs/kit` to specify the HTTP status code and an optional message:
+
+```js
+/// file: src/routes/admin/+layout.server.js
+// @filename: ambient.d.ts
+declare namespace App {
+	interface Locals {
+		user: {
+			name: string;
+			isAdmin: boolean;
+		}
+	}
+}
+
+// @filename: $types.d.ts
+export type LayoutServerLoad = import('@sveltejs/kit').ServerLoad<{}>;
+
+// @filename: index.js
+// ---cut---
+import { error } from '@sveltejs/kit';
+
+/** @type {import('./$types').LayoutServerLoad} */
+export function load({ locals }) {
+	if (!locals.user) {
+		throw error(401, 'not logged in');
+	}
+
+	if (!locals.user.isAdmin) {
+		throw error(403, 'not an admin');
+	}
+}
+```
+
+If an _unexpected_ error is thrown, SvelteKit will invoke [`handleError`](/docs/hooks#handleerror) and treat it as a 500 Internal Server Error.
+
+> In development, stack traces for unexpected errors are visible as `$page.error.stack`. In production, stack traces are hidden.
+
+### Redirects
+
+Redirecting users works similarly:
```

Comment:
This is more of a personal preference, but I prefer to have the sections more self-contained. If I jump right into this section I'm lost what this "similarly" refers to.

```suggestion
To redirect users, use the `redirect` helper from `@sveltejs/kit` to specify the location to which they should be redirected alongside a `3xx` status code.
```

Reply:
With the beautiful Cmd+K search UI from the docs, I think it is important to keep every section self contained. If I look for 'redirects' I would like to only need to read that section, and not read back to what _similarly_ means. Your suggestion clearly says what I should do when I want to redirect an user. 

TL;DR: great suggestion 👍🏻